### ISSUE TITLE
Creating a TKG cluster in VCD v10.5 get stuck 

### ISSUE THREAD

**Issue by josajunior81** [vmware/terraform-provider-vcd#1118]
I've cloned the example of how to create a TKG cluster with terraform provider and changed the tfvars with mine. 

I'm using an API Token user to login and create the cluster, this user is Organization Administrator. With this user I can create a Cluster via VCD UI.

The terraform runs OK
```shell
terraform apply ./plan.tfplan
vcd_rde.k8s_cluster_instance: Creating...
vcd_rde.k8s_cluster_instance: Creation complete after 1s [id=urn:vcloud:entity:vmware:capvcdCluster:d01ae7f2-d47a-49c9-a672-bd2393a96642]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

But the creation get stuck after "CapiYamlProcessed" goes OK. After this event nothing more happens. 

![image](https://github.com/vmware/terraform-provider-vcd/assets/287938/62e58ad0-4636-4e8f-ba95-274c8bcea699)

**Comment by adambarreiro** [comment]
Hi @josajunior81, thanks for reaching out.

Did you also install the Container Service Extension v4.0 Server [with Terraform](https://registry.terraform.io/providers/vmware/vcd/latest/docs/guides/container_service_extension_4_0_install), or was it there already an existing installation in place? Is it running correctly?

You can check the following:

- See the logs in the virtual machine that runs the VMware Cloud Director Container Service Extension 4.0 OVA file, they're located at **/root/cse.log**, there it is displayed the possible errors with the processing of the K8s cluster request. If other K8s clusters requests are OK, I doubt there are errors here, but just to discard it.

- If there are no errors, you could check whether the EPHEMERAL_VM that bootstraps the clusters is created (and able to Power ON) or not (if it's created, one can also check the logs)

- You can try to check the output of the `vcd_rde`, this is in the `computed_entity` attribute. This is what CSE returns on K8s cluster creation requests, and it is updated on every refresh with the cluster information. There are some useful fields there that show possible errors during the creation. You can decode the whole JSON and do some lookups on the fields, for example:

```
locals {
  k8s_cluster_computed = jsondecode(vcd_rde.k8s_cluster_instance.computed_entity)
  has_status           = lookup(local.k8s_cluster_computed, "status", null) 
}

output "computed_k8s_cluster_events" {
  value = local.has_status ? local.k8s_cluster_computed["status"]["vcdKe"]["eventSet"] : null
}
```

This output shows the events during cluster creation, which are located in the JSON at `.status.vcdKe.eventSet` (it is an array of error/debug messages).

Let's see whether we can spot the issue with this information, it could be that the CAPVCD Yaml is malformed for some reason, but that's only a rough guess.

**Comment by josajunior81** [comment]
> Hi @josajunior81, thanks for reaching out.
> 
> Did you also install the Container Service Extension v4.0 Server [with Terraform](https://registry.terraform.io/providers/vmware/vcd/latest/docs/guides/container_service_extension_4_0_install), or was it there already an existing installation in place? Is it running correctly?
> 
> You can check the following:
> 
> * See the logs in the virtual machine that runs the VMware Cloud Director Container Service Extension 4.0 OVA file, they're located at **/root/cse.log**, there it is displayed the possible errors with the processing of the K8s cluster request. If other K8s clusters requests are OK, I doubt there are errors here, but just to discard it.
> * If there are no errors, you could check whether the EPHEMERAL_VM that bootstraps the clusters is created (and able to Power ON) or not (if it's created, one can also check the logs)
> * You can try to check the output of the `vcd_rde`, this is in the `computed_entity` attribute. This is what CSE returns on K8s cluster creation requests, and it is updated on every refresh with the cluster information. There are some useful fields there that show possible errors during the creation. You can decode the whole JSON and do some lookups on the fields, for example:
> 
> ```
> locals {
>   k8s_cluster_computed = jsondecode(vcd_rde.k8s_cluster_instance.computed_entity)
>   has_status           = lookup(local.k8s_cluster_computed, "status", null) 
> }
> 
> output "computed_k8s_cluster_events" {
>   value = local.has_status ? local.k8s_cluster_computed["status"]["vcdKe"]["eventSet"] : null
> }
> ```
> 
> This output shows the events during cluster creation, which are located in the JSON at `.status.vcdKe.eventSet` (it is an array of error/debug messages).
> 
> Let's see whether we can spot the issue with this information, it could be that the CAPVCD Yaml is malformed for some reason, but that's only a rough guess.

The CSE was not installed with Terraform and the version is 4.1, could be it?

Theres no errors in the cse.log and no EPHEMERAL_VMs created.

The ko8s_cluster_computed output is:

```hcl
Outputs:

computed_k8s_cluster_events = [
  {
    "additionalDetails" = {
      "Detailed Event" = "Successfully connected to vcdclient[org-serpro-supse:josafassj] for cluster [tkg-terraform(urn:vcloud:entity:vmware:capvcdCluster:c7e68269-4a58-466b-8b2e-37d350826b86)] and acquired the RDE"
    }
    "name" = "VcdClientAuthorized"
    "occurredAt" = "2023-09-11T13:50:23.255650945Z"
  },
  {
    "additionalDetails" = {
      "Detailed Event" = "Successfully processed CapiYaml for cluster: [tkg-terraform(urn:vcloud:entity:vmware:capvcdCluster:c7e68269-4a58-466b-8b2e-37d350826b86)]"
    }
    "name" = "CapiYamlProcessed"
    "occurredAt" = "2023-09-11T13:50:24.537831391Z"
    "vcdResourceId" = "urn:vcloud:entity:vmware:capvcdCluster:c7e68269-4a58-466b-8b2e-37d350826b86"
    "vcdResourceName" = "tkg-terraform"
  },
]
computed_k8s_cluster_status = "provisioning"
```

**Comment by adambarreiro** [comment]
> The CSE was not installed with Terraform and the version is 4.1, could be it?

I'm afraid yes.

Currently the RDE schemas and payloads are for CSE v4.0. We're working to update the guide to have the new required files and resources for CSE v4.1.

**Comment by josajunior81** [comment]
> Currently the RDE schemas and payloads are for CSE v4.0. We're working to update the guide to have the new required files and resources for CSE v4.1.

Ok I'll wait the update and do the tests again, thanks!

**Comment by adambarreiro** [comment]
Hi @josajunior81,

Although it is not released yet, we merged the required changes for CSE v4.1 to work. This is the new content:
- Installation guide updated to v4.1: https://github.com/vmware/terraform-provider-vcd/blob/main/website/docs/guides/container_service_extension_4_x_install.html.markdown
- Cluster management guide updated to v4.1: https://github.com/vmware/terraform-provider-vcd/blob/main/website/docs/guides/container_service_extension_4_x_cluster_management.html.markdown

It requires latest VCD Terraform Provider, as for now (as it is not released) it can be installed by cloning this repo and running `make install`. Not sure whether it would be possible for you to try it out? Any feedback would be great!

Thanks in advance.
